src/models/linePlusBarChart.js

```diff
@@ -600,6 +627,13 @@ nv.models.linePlusBarChart = function() {
         }}
     });
 
+    chart.toggleAxis = function() {
+        switchYAxisOrder = !switchYAxisOrder;
+        var tickFormat = y1Axis.tickFormat();
+        y1Axis.tickFormat(y2Axis.tickFormat());
+        y2Axis.tickFormat(tickFormat);
+    }
+
```

Comment:
I would do option 3, inside the setter...  that seems like the cleanest thing since that option drives the behavior.